samza-core/src/main/java/org/apache/samza/storage/StorageRecovery.java

```diff
@@ -207,7 +188,9 @@ private void getChangeLogMaxPartitionNumber() {
    */
   @SuppressWarnings({ "unchecked", "rawtypes" })
   private void getTaskStorageManagers() {
-    StreamMetadataCache streamMetadataCache = new StreamMetadataCache(Util.javaMapAsScalaMap(systemAdmins), 5000, SystemClock.instance());
+    Map<String, SystemAdmin> systemAdminMap = new JavaSystemConfig(jobConfig).getSystemAdmins();
+    StreamMetadataCache streamMetadataCache = new StreamMetadataCache(Util.javaMapAsScalaMap(systemAdminMap), 5000, SystemClock.instance());
+    systemAdmins = new SystemAdmins(Util.javaMapAsScalaMap(systemAdminMap));
```

Comment:
Seems weird that this is set as a side effect in some unrelated method. Can this happen in the constructor?

Reply:
Good point. I have updated the patch to instantiate SystemAdmin in the constructor.